### PR TITLE
wn/chute

### PR DESCRIPTION
**Pull request by selym3**
To Mr. Pratham,

It would be greatly appreciated if you could take the time out of your day to analyze this branch.

Sincerely,
Myles

### CODE REVIEW

File: src/main/java/com/stuypulse/robot/Constants.java [hunk 1]
```diff
@@ -59,4 +59,13 @@
...
     int INTAKE_MOTOR_PORT = -1;
     int INTAKE_SOLENOID_PORT = -1;
+
+    /*********************************************************************************************
+     * Chute Constants
+     *********************************************************************************************/
+	int CHUTE_LIFT_MOTOR_PORT = -1;
+	int CHUTE_LOWER_SENSOR_PORT = -1;
+	int CHUTE_UPPER_SENSOR_PORT = -1;
+	double CHUTE_LIFT_UP_SPEED = -1;
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/Constants.java]: Separate ports and constants into two different sections.

File: src/main/java/com/stuypulse/robot/commands/ChuteDownCommand.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public void execute() {
+        chute.liftDown();
+    }
+
+    @Override
+    public boolean isFinished() {  
+        return chute.getLowerChuteValue();
+    }
+
+    /*@Override
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/commands/ChuteDownCommand.java]: Why is this commented out?

File: src/main/java/com/stuypulse/robot/commands/ChuteIntakeCommand.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    }
+
+    @Override
+    public boolean isFinished() {  
+        return false;
+    }
+
+    /*@Override
+    public void end(boolean interrupted) {
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/commands/ChuteIntakeCommand.java]: Why is this commented out?

File: src/main/java/com/stuypulse/robot/commands/ChuteUpCommand.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Override
+    public boolean isFinished() {  
+        return chute.getUpperChuteValue();
+    }
+
+    /*@Override
+    public void end(boolean interrupted) {
+        chute.stopChute();
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/commands/ChuteUpCommand.java]: Why is this commented out too?

File: src/main/java/com/stuypulse/robot/subsystems/Chute.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+    public void liftDown() {
+        chuteMotor.set(-Constants.CHUTE_LIFT_UP_SPEED);
+    }
+
+    public void stopChute() {
+        chuteMotor.stopMotor();
+    }
+
+    public double getTicks() {
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/subsystems/Chute.java]: When do you use these three methods?

File: vendordeps/Phoenix.json [hunk 6]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (rmui1) [inline vendordeps/Phoenix.json:1]: When do you need this?

File: vendordeps/navx_frc.json [hunk 7]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (rmui1) [inline vendordeps/navx_frc.json]: When do you need this?

File: src/main/java/com/stuypulse/robot/commands/ChuteIntakeCommand.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    @Override
+    public void initialize() {
+
+    }
+
+    @Override
+    public void execute() {
+        if (previousValue != RobotContainer.gamepad.getRawDPadDown()) {
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/commands/ChuteIntakeCommand.java]: Why do you need this part? OI (configuring commands to buttons) should already do this for you

File: src/main/java/com/stuypulse/robot/commands/ChuteIntakeCommand.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            isIntake = !isIntake;
+            previousValue = RobotContainer.gamepad.getRawDPadDown();
+        } 
+
+        if (isIntake) {
+            if (chute.getLowerChuteValue()) {
+                chute.liftUp();
+            } else {
+                chute.stopChute();
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/commands/ChuteIntakeCommand.java]: You don't need this here; just put it in end.

File: src/main/java/com/stuypulse/robot/commands/ChuteUpCommand.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+import com.stuypulse.robot.subsystems.Chute;
+
+public class ChuteUpCommand extends CommandBase {
+
+    private Chute chute;
+
+    public ChuteUpCommand(Chute chute) {
+        this.chute = chute;
+
+        // Use addRequirements() here to declare subsystem dependencies.
```
Comment (rmui1) [inline src/main/java/com/stuypulse/robot/commands/ChuteUpCommand.java]: Delete this comment

**Review by rmui1 — DISMISSED** [review]
Oops, forgot to mention this before, but merge master into your branch and fix the merge conflicts

**Comment by PrathamRawat** [comment]
I disagree with some of Renee’s comments, we’ll talk about it in person

File: src/main/java/com/stuypulse/robot/commands/ChimneyStopCommand.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
+package com.stuypulse.robot.commands;
+
+import com.stuypulse.robot.subsystems.Chimney;
+import com.stuypulse.stuylib.input.Gamepad;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public class ChimneyStopCommand extends CommandBase {
```
Comment (PrathamRawat) [inline src/main/java/com/stuypulse/robot/commands/ChimneyStopCommand.java:8]: Should be an instant command

File: src/main/java/com/stuypulse/robot/utils/NeoEncoder.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
+package com.stuypulse.robot.utils;
+/* Import this from stuylib */
+
+import com.revrobotics.CANEncoder;
+
+public final class NeoEncoder {
```
Comment (PrathamRawat) [inline src/main/java/com/stuypulse/robot/utils/NeoEncoder.java]: This file should be removed

**Review by PrathamRawat — CHANGES_REQUESTED** [review]

File: src/main/java/com/stuypulse/robot/commands/ChimneyStopCommand.java (hunk 11 above)
Comment (selym3) [reply src/main/java/com/stuypulse/robot/commands/ChimneyStopCommand.java:8]: Review in code review 2

**Review by PrathamRawat — APPROVED** [review]
Good work guys

**Pull request merged**